src/events/mod.rs

```diff
@@ -170,6 +170,22 @@ impl<'a> BytesStart<'a> {
         bytes.extend_from_slice(&*a.value);
         bytes.push(b'"');
     }
+
+    /// Edit the name of the BytesStart in-place
+    pub fn set_name(&mut self, name: &[u8]) -> &mut BytesStart<'a> {
+        {
+            let bytes = self.buf.to_mut();
+            bytes.splice(..self.name_len, name.to_vec());
```

Comment:
@meven  I don't think the `to_vec` call here is needed. It'd be good to avoid the extra allocation!

Reply:
I did not find a better way I would expect the same but I ran into 
```console
error[E0271]: type mismatch resolving `<&[u8] as std::iter::IntoIterator>::Item == u8`
   --> src/events/mod.rs:178:19
    |
178 |             bytes.splice(..self.name_len, name);
    |                   ^^^^^^ expected reference, found u8
    |
    = note: expected type `&u8`
               found type `u8`
```
It seems I still need some conversion, name `&[u8]` implements `IntoIterator<u8>` but I need an `IntoIterator<&u8>`. Maybe There is some simple trick I am missing.

Reply:
Ah, I've run into this in the past. It's a little silly. Basically what you want is `name.iter().cloned()` or `name.iter().map(|&b| b)` to tell the compiler that you want to utilize the fact that `u8` is `Copy`.

Reply:
Thanks, will make another PR